### ISSUE TITLE
Text Decoration Property

### ISSUE THREAD

**Issue by davidseik** [airbnb/react-sketchapp#142]
| -------------------------------------------------------------------------------------------------
| Reporting a bug or issue
| -------------------------------------------------------------------------------------------------

Hi, i'm trying to use the textDecoration property to set underline to a text contained in a button.
The button is made using a `<Text />` element with the appropriate styles. First I tried to add the property to the root like this `<Text  style={{ textDecoration: 'underline' }}>some text</Text>` but it didn't work. Proceeded to separate the styles and try to style directly the children but no luck.

I also couldn't see the support of the property in here:
http://airbnb.io/react-sketchapp/docs/styling.html

**Expected behavior:**

![screen shot 2017-07-06 at 10 34 16 am](https://user-images.githubusercontent.com/4756314/27924597-bbd099b2-6236-11e7-9e08-2e9746b2483e.png)

**Observed behavior:**

![screen shot 2017-07-06 at 10 34 28 am](https://user-images.githubusercontent.com/4756314/27924614-c40c06d4-6236-11e7-8926-9666c7c33af2.png)

**How to reproduce:**
```js
const Button = ({ text, style }) => {
  return (
    <Text
      style={style}
      name="container"
    >
      <Text style={{ textDecoration: 'underline'}}>
        {text}
      </Text>
    </Text>
  );
};
```

**Sketch version:**
44.1

**Comment by jongold** [comment]
Thanks for the report!

Looks like React Native has a slightly different syntax for [text decoration](https://facebook.github.io/react-native/docs/text.html#style) — `textDecorationLine`, `textDecorationColor`, `textDecorationStyle`. My guess is that these haven't been implemented yet either, but thanks for alerting us to it 🙏

**Comment by davidseik** [comment]
@jongold is there any place I could start looking into this? I would like to contribute but honestly I don't know where to start over this. Let me know if you have any thoughts around this.

**Comment by jongold** [comment]
for sure!

basic workflow I'd suggest:
- save & close any sketch files you have open!
- install [`copy-sketch-json`](https://github.com/jongold/copy-sketch-json)
- make a barebones sketch file just with one text layer w/ underline
- use that plugin to copy the JSON to your text editor (or a JSON viewer); dig through it to see what JSON you need to generate for an underline
- clone the main repo, if you don't have it already, and npm install all the dependencies etc
- poke around [`src/jsonUtils/textLayers.js`](https://github.com/airbnb/react-sketchapp/blob/master/src/jsonUtils/textLayers.js) to see the corresponding portion of JSON, make the changes
- to test it out, in the root of react-sketchapp `npm link` and then `npm run watch`
- open a new terminal, go into one of the `examples/` folders, `npm install`, `npm link react-sketchapp`, `npm run render`
- open that example in your editor and add the corresponding react code — `<Text style={{textDecorationLine: 'underline'}}…`
- see if it works / crashes Sketch / etc, rinse, repeat :D

**Comment by davidseik** [comment]
Hi Jon, I was able to install the plugin on my sketch but upon running i'm not getting the results on my clipboard.

I checked my the OS's console and I see this error:
```
7/6/17 2:24:40.127 PM Copy as JSON (Sketch)[81363]: SyntaxError: Unexpected token '>'.
Plugin “Copy as JSON”, line 98:
» const copyToClipboard = text => {
```

Seems like sketch is not able to execute this command or something; I initially thought it was a problem of my node version (as I was using V6) but I upgraded to the latest V8 and no issues. I know it's a side-track. Should I open the issue on your copy-sketch-json for discussion?
